### PR TITLE
HAWQ-1103. Send constant datatype and length in filter to PXF.

### PR DESCRIPTION
**Pull request by sansanichfb**

### CODE REVIEW

File: src/backend/access/external/test/pxffilters_test.c [hunk 1]
```diff
@@ -514,23 +514,26 @@ test__pxf_serialize_filter_list__manyFilters(void **state)
...
 
 	expressionItems = lappend(expressionItems, expressionItem1);
 	expressionItems = lappend(expressionItems, expressionItem2);
 	expressionItems = lappend(expressionItems, expressionItem3);
 	expressionItems = lappend(expressionItems, expressionItem4);
+	expressionItems = lappend(expressionItems, expressionItem5);
 
 	result = pxf_serialize_filter_list(expressionItems);
-	assert_string_equal(result, "a0c\\\"1984\\\"o5a1c\\\"\"George Orwell\"\\\"o5a2c\\\"\"Winston\"\\\"o5a3c\\\"\"Eric-%\"\\\"o7");
+	assert_string_equal(result, "a0c25s4d1984o5a1c25s13dGeorge Orwello5a2c25s7dWinstono5a3c25s6dEric-%o7a4c25s25d\"Ugly\" string with quoteso5");
```
Comment (sansanichfb) [inline src/backend/access/external/test/pxffilters_test.c]: Sure, will add it

File: src/include/access/pxffilters.h [hunk 2]
```diff
@@ -53,9 +53,11 @@ typedef enum PxfOperatorCode
...
  */
-#define PXF_ATTR_CODE		'a'
-#define PXF_CONST_CODE		'c'
-#define PXF_OPERATOR_CODE	'o'
+#define PXF_ATTR_CODE				'a'
+#define PXF_CONST_CODE				'c'
+#define PXF_SIZE_BYTES				's'
+#define PXF_CONST_DATA				'd'
+#define PXF_OPERATOR_CODE			'o'
 #define PXF_LOGICAL_OPERATOR_CODE	'l'
```
Comment (sansanichfb) [inline src/include/access/pxffilters.h]: It looks aligned in IDE.

File: src/backend/access/external/pxffilters.c [hunk 3]
```diff
@@ -330,7 +329,27 @@ pxf_free_filter(PxfFilterDesc* filter)
  *
  * Yields the following serialized string:
  *
- * a0c1o2a0c5o1o7a2c"third"o5o7
+ * a0c23s1d1o2a1c23s1d5o1a2c25s5dthirdo5l0l0
```
Comment (shivzone) [inline src/backend/access/external/pxffilters.c]: Update above comment Each filter is serialized as follows:
<attcode><attnum><constcode><constval><constsizecode><constsize><constdata><constvalue><opercode><opernum>

File: src/backend/access/external/test/pxffilters_test.c [hunk 4]
```diff
@@ -514,23 +514,28 @@ test__pxf_serialize_filter_list__manyFilters(void **state)
...
 
 	ExpressionItem* expressionItem1 = build_expression_item(1, TEXTOID, "1984", TEXTOID, TextEqualOperator);
-	ExpressionItem* expressionItem2 = build_expression_item(2, TEXTOID, "\"George Orwell\"", TEXTOID, TextEqualOperator);
-	ExpressionItem* expressionItem3 = build_expression_item(3, TEXTOID, "\"Winston\"", TEXTOID, TextEqualOperator);
-	ExpressionItem* expressionItem4 = build_expression_item(4, TEXTOID, "\"Eric-%\"", TEXTOID, 1209);
+	ExpressionItem* expressionItem2 = build_expression_item(2, TEXTOID, "George Orwell", TEXTOID, TextEqualOperator);
+	ExpressionItem* expressionItem3 = build_expression_item(3, TEXTOID, "Winston", TEXTOID, TextEqualOperator);
+	ExpressionItem* expressionItem4 = build_expression_item(4, TEXTOID, "Eric-%", TEXTOID, 1209);
+	ExpressionItem* expressionItem5 = build_expression_item(5, TEXTOID, "\"Ugly\" string with quotes", TEXTOID, TextEqualOperator);
+	ExpressionItem* expressionItem6 = build_expression_item(6, TEXTOID, "", TEXTOID, TextEqualOperator);
```
Comment (shivzone) [inline src/backend/access/external/test/pxffilters_test.c]: Add a test for empty byte array as well
